indico/modules/events/contributions/util.py

```diff
@@ -189,3 +196,50 @@ def get_contribution_ical_file(contrib):
     data = {'results': serialize_contribution_for_ical(contrib)}
     serializer = Serializer.create('ics')
     return BytesIO(serializer(data))
+
+
+def import_contributions_from_csv(event, f):
+    """Import timetable contributions from a CSV file into an event."""
+    reader = csv.reader(f)
+    contributions = []
+    for n_row, row in enumerate(reader):
+        try:
+            start_dt, duration, title, first_name, last_name, affiliation, email = row
+        except ValueError:
+            raise UserValueError(_('Row {}: malformed CSV data - please check that the number of columns is correct')
+                                 .format(n_row))
+        try:
+            parsed_start_dt = event.tzinfo.localize(dateutil.parser.parse(start_dt)) if start_dt else None
+        except ValueError:
+            raise UserValueError(_("Row {}: can't parse date: \"{}\"").format(n_row, start_dt))
+
+        if parsed_start_dt and not duration:
+            raise UserValueError(_("Row {}: a start date/time requires a duration").format(n_row))
+
+        try:
+            parsed_duration = timedelta(minutes=int(duration)) if duration else None
+        except ValueError:
+            raise UserValueError(_("Row {}: can't parse duration: {}").format(n_row, duration))
+
+        with track_time_changes() as changes:
+            contribution = create_contribution(event, {
+                'start_dt': parsed_start_dt,
+                'duration': parsed_duration or timedelta(minutes=20),
+                'title': title
+            }, extend_parent=True)
+
+        if not email:
+            continue
+
+        # set the information of the speaker
+        person = get_event_person(event, {
+            'firstName': first_name,
+            'familyName': last_name,
+            'affiliation': affiliation,
+            'email': email
+        })
+        link = ContributionPersonLink(person=person, is_speaker=True)
+        contribution.person_links.append(link)
+        contributions.append(contribution)
+
+    return contributions, changes
```

Comment:
you return the `changes` from the last iteration but nothing from earlier.. maybe better to set a flag within the loop if there was a change and return that instead?